cmd/influx_inspect/export/export.go

```diff
@@ -0,0 +1,395 @@
+	err := filepath.Walk(cmd.dataDir, func(dir string, f os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// check to see if this is a tsm file
+		ext := fmt.Sprintf(".%s", tsm1.TSMFileExtension)
+		if filepath.Ext(dir) != ext {
+			return nil
+		}
+
+		relPath, _ := filepath.Rel(cmd.dataDir, dir)
+		dirs := strings.Split(relPath, string(byte(os.PathSeparator)))
+		if len(dirs) < 2 {
+			return fmt.Errorf("invalid directory structure for %s", dir)
+		}
+		if dirs[0] == cmd.database || cmd.database == "" {
+			if dirs[1] == cmd.retentionPolicy || cmd.retentionPolicy == "" {
+				key := filepath.Join(dirs[0], dirs[1])
+				files := cmd.tsmFiles[key]
+				if files == nil {
+					files = []string{}
+				}
+				cmd.manifest[key] = struct{}{}
+				cmd.tsmFiles[key] = append(files, dir)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cmd *Command) walkWALFiles() error {
+	err := filepath.Walk(cmd.walDir, func(dir string, f os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// check to see if this is a wal file
+		prefix := tsm1.WALFilePrefix
+		ext := fmt.Sprintf(".%s", tsm1.WALFileExtension)
+		_, fileName := path.Split(dir)
+		if filepath.Ext(dir) != ext || !strings.HasPrefix(fileName, prefix) {
+			return nil
+		}
+
+		relPath, _ := filepath.Rel(cmd.walDir, dir)
+		dirs := strings.Split(relPath, string(byte(os.PathSeparator)))
+		if len(dirs) < 2 {
+			return fmt.Errorf("invalid directory structure for %s", dir)
+		}
+		if dirs[0] == cmd.database || cmd.database == "" {
+			if dirs[1] == cmd.retentionPolicy || cmd.retentionPolicy == "" {
+				key := filepath.Join(dirs[0], dirs[1])
+				files := cmd.walFiles[key]
+				if files == nil {
+					files = []string{}
+				}
+				cmd.manifest[key] = struct{}{}
+				cmd.walFiles[key] = append(files, dir)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cmd *Command) writeFiles() error {
+	// open our output file and create an output buffer
+	var w io.WriteCloser
+	w, err := os.Create(cmd.out)
+	if err != nil {
+		return err
+	}
+	defer w.Close()
+	if cmd.compress {
+		w = gzip.NewWriter(w)
+		defer w.Close()
+	}
+
+	s, e := time.Unix(0, cmd.startTime).Format(time.RFC3339), time.Unix(0, cmd.endTime).Format(time.RFC3339)
+	fmt.Fprintf(w, "# INFLUXDB EXPORT: %s - %s\n", s, e)
+
+	// Write out all the DDL
+	fmt.Fprintln(w, "# DDL")
+	for key := range cmd.manifest {
+		keys := strings.Split(key, string(byte(os.PathSeparator)))
+		db, rp := influxql.QuoteIdent(keys[0]), influxql.QuoteIdent(keys[1])
+		fmt.Fprintf(w, "CREATE DATABASE %s WITH NAME %s\n", db, rp)
+	}
+
+	fmt.Fprintln(w, "# DML")
+	for key := range cmd.manifest {
+		keys := strings.Split(key, string(byte(os.PathSeparator)))
+		fmt.Fprintf(w, "# CONTEXT-DATABASE:%s\n", keys[0])
+		fmt.Fprintf(w, "# CONTEXT-RETENTION-POLICY:%s\n", keys[1])
+		if files, ok := cmd.tsmFiles[key]; ok {
```

Comment:
the wal files are, but the TSM are not, I will update that to use a sorted list for the tsm files as well.